plugins/fastem_set_factory_values.py

```diff
@@ -0,0 +1,97 @@
+
+@author: Thera Pals
+
+During debugging or service actions it can be necessary to be able to load the
+factory settings of the descanner. This will make sure the pattern is visible on
+the diagnostic camera. This plugin adds an option in the help dropdown that will
+show a pop-up before loading the factory settings.
+
+Copyright © 2024 Thera Pals, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+"""
+
+import logging
+
+import wx
+
+from odemis import model
+from odemis.gui.plugin import Plugin
+
+
+class SetFactoryValues(Plugin):
+    name = "Set factory values"
+    __version__ = "1.0"
+    __author__ = "Thera Pals"
+    __license__ = "GPLv2"
+
+    def __init__(self, microscope, main_app):
+        super().__init__(microscope, main_app)
+
+        # It only makes sense if the FASTEM acquisition tab is present
+        try:
+            self.main_frame = main_app.main_frame
+            fastem_main_tab = main_app.main_data.getTabByName("fastem_main")
+            main_data = main_app.tab_controller.main_data
+            self.descanner = main_data.descanner
+            self.stage = main_data.stage
+            self.mppc = main_data.mppc
+            self.ccd = main_data.ccd
+            self._acquisition_tab = fastem_main_tab.acquisition_tab
+        except LookupError:
+            logging.debug(
+                "Not loading set-factory-values tool since acquisition tab is not present."
+            )
+            return
+
+        self._acquisition_controller = self._acquisition_tab._acquisition_controller
+
+        self.addMenu("Help/Development/Set factory values",
+                     self._set_factory_values,
+                     item_kind=wx.ITEM_NORMAL,
+                     pass_menu_item=False)
+
+    def _set_factory_values(self):
+        """
+        Menu callback for: Help/Development/Set factory values. Shows a pop-up to load factory settings.
+        If yes is clicked, the pop-up remains open until the factory settings are uploaded (~4 seconds).
+        """
+        box = wx.MessageDialog(
+            self.main_frame,
+            "Do you want to load factory settings for descanner offset and amplitude,"
+            " and set the z-stage to the favorite position?",
+            style=wx.YES_NO | wx.ICON_QUESTION | wx.CENTER,
+        )
+        ans = box.ShowModal()  # Waits for the window to be closed
+        if ans == wx.ID_YES:  # only set values when "yes" has been selected
```

Comment:
It's usually nicer to read the "normal" code at the lowest indentation. So I'd inverse the if like that: 
```suggestion
        if ans != wx.ID_NO:  # only set values when "yes" has been selected
           return
```

Reply:
I might be getting lost in the double negation, but I think your suggestion should be 
```
        if ans == wx.ID_NO:  # only set values when "yes" has been selected
           return
```

Reply:
Yes, indeed!

Reply:
Actually, I think the safest is `if ans != wx.ID_YES` (because maybe it can also return values like ID_CLOSE, etc?)